<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>

<!-- Mobile Redirect -->
<script src="http://static.dudamobile.com/DM_redirect.js" type="text/javascript"></script>
<script type="text/javascript">DM_redirect("http://m.bioinformaticsalgorithms.com");</script>

<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<title>FAQs | Chapter 4 | How Do We Sequence Antibiotics? </title>
<link rel="stylesheet" type="/text/css" href="../Styles/960_24_col.css" />
<link rel="stylesheet" type="/text/css" href="../Styles/reset.css" />
<link rel="stylesheet" type="/text/css" href="../Styles/text.css" />


<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-51449831-1', 'bioinformaticsalgorithms.com');
  ga('send', 'pageview');

</script>

<!-- Needed for JQuery in order to have expandable FAQs -->
<script type="text/javascript" src="//code.jquery.com/jquery-1.11.0.min.js"></script>

<!-- JQuery Javascript code -->
<script>
 $(document).ready(function() {

	$('.faq_question').click(function() {

		if ($(this).parent().is('.open')){
			$(this).closest('.faq').find('.faq_answer_container').animate({'height':'0'},500);
			$(this).closest('.faq').removeClass('open');

			}else{
				var newHeight =$(this).closest('.faq').find('.faq_answer').height() +'px';
				$(this).closest('.faq').find('.faq_answer_container').animate({'height':newHeight},500);
				$(this).closest('.faq').addClass('open');
			}

	});

});
</script>

</head>

<body>

    <div id="content" class="container_24 clearfix">
      <div class="main-content">
        <div class="header-img">
          <img src="../images/header-img.jpg" alt="Bioinformatics Algorithms: An Active Learning Approach" width="960" height="330" />
        </div>
        <div class="nav">
          <ul>
            <li><a href="../index.htm">Home</a></li>
            <li><a href="../about-the-author.htm">Authors</a></li>
            <li><a href="../contents.htm">Contents</a></li>
            <li><a href="../faqs.htm">FAQs</a></li>
            <li><a href="../videos.htm">Videos & Slides</a></li>
                        <li><a href="https://secure.mybookorders.com/Orderpage/1402">Buy the Book!</a></li>
            <li><a class="active" href="../contact.htm">Contact</a></li>
          </ul>
        </div>

    	<div class="section-about">

<h1>Chapter 4: How Do We Sequence Antibiotics?</h1>

<h3><a name="week3"></a>(Coursera Week 3)</h3>




<div class="faq_container">
	<div class="faq">
		<div class="faq_question">How do I expand the list of candidate peptides in <b>CyclopeptideSequencing</b>?</div>
		<div class="faq_answer_container">
			<div class="faq_answer">
				<p>One of the students in the first session of our class, John Cloutier, provided the following example that illustrates how <strong>CyclopeptideSequencing</strong> works.  Consider a strange amino acid alphabet consisting of just two amino acids with masses 1 and 3.  The figure below shows the peptides generated at each step by <strong>CyclopeptideSequencing</strong> with respect to a sample experimental spectrum {0, 1, 1, 1, 2, 2, 3, 3, 4, 4, 5, 5, 5, 6}.  Consistent peptides are shown in black, and inconsistent peptides are shown in red.  In step 4, <strong>CyclopeptideSequencing</strong> produces the blue peptides 1-1-1-3, 1-1-3-1, 1-3-1-1, and 3-1-1-1; these four linear peptides all represent the same cyclic peptide, whose spectrum is equal to the experimental spectrum.</p>

				<p><br></p>

				<center>
				<table>
				<col width="75">
				<col width="190">
				<col width="310">
				<tbody>
				<tr><th><b>Step</b></th><th><b>Candidate peptide</b></th><th><b>Linear Spectrum</b></th></tr>

<tr><td>1</td><td>(1) </td><td>(0, 1) </td></tr>

<tr><td></td><td>(3)</td><td>(0, 3) </td></tr>

<tr><td></td><td></td><td></td></tr>

<tr><td>&nbsp;</td></tr>

<tr><td>2</td><td>(1, 1)    </td><td>    (0, 1, 1, 2) </td></tr>

<tr><td></td><td>(1, 3)</td><td>  (0, 1, 3, 4)        </td></tr>

<tr><td></td><td>(3, 1)</td><td>  (0, 1, 3, 4) </td></tr>

<tr><td></td><td>(3, 3)</td><td>(0, 3, 3, 6) </td></tr>

<tr><td>&nbsp;</td></tr>

<tr><td>3</td><td> (1, 1, 1)            </td><td>(0, 1, 1, 1, 2, 2, 3) </td></tr>

<tr><td></td><td>(1, 1, 3)       </td><td> (0, 1, 1, 2, 3, 4, 5) </td></tr>

<tr><td></td><td> (1, 3, 1)   </td><td>  (0, 1, 1, 3, 4, 4, 5) </td></tr>

<tr><td></td><td><font color="red">(1, 3, 3)  </font></td><td><font color="red"> (0, 1, 3, 3, 4, 6, 7) - remove</font></td></tr>

<tr><td></td><td>(3, 1, 1)   </td><td>     (0, 1, 1, 2, 3, 4, 5) </td></tr>

<tr><td></td><td> <font color="red">(3, 1, 3)</font>      </td><td><font color="red"> (0, 1, 3, 3, 4, 4, 7) - remove</font></td></tr>

<tr><td></td><td><font color="red"> (3, 3, 1)    </font></td><td><font color="red">   (0, 1, 3, 3, 4, 6, 7) - remove</font></td></tr>

<tr><td></td><td>  <font color="red">(3, 3, 3)    </font></td><td>   <font color="red">(0, 3, 3, 3, 6, 6, 9) - remove</font></td></tr>

<tr><td>&nbsp;</td></tr>

<tr><td>4</td><td> <font color="red">(1, 1, 1, 1)      </font></td><td><font color="red"> (0, 1, 1, 1, 1, 2, 2, 2, 3, 3, 4) - remove</font></td></tr>

<tr><td></td><td><font color="blue">  (1, 1, 1, 3)       </font></td><td> <font color="blue">(0, 1, 1, 1, 2, 2, 3, 3, 4, 5, 6) – output</font></td></tr>

<tr><td></td><td>     <font color="blue">(1,1, 3, 1)          </font></td><td><font color="blue">  (0, 1, 1, 1, 2, 3, 4, 4, 5, 5, 6) – output </font></td></tr>

<tr><td></td><td>      <font color="red">(1, 1, 3, 3)    </font></td><td><font color="red">  (0, 1, 1, 2, 3, 3, 4, 6, 5, 7, 8) - remove</font></td></tr>

<tr><td></td><td><font color="blue">       (1, 3, 1, 1)   </font></td><td> <font color="blue">(0, 1, 1, 1, 2, 3, 4, 4, 5, 5, 6) - output </font></td></tr>

<tr><td></td><td> <font color="red">(1, 3, 1, 3)   </font></td><td><font color="red">(0, 1, 1, 2, 3, 3, 4, 6, 5, 7, 8)  - remove</font></td></tr>

<tr><td></td><td><font color="blue">(3, 1, 1, 1)    </font></td><td><font color="blue">(0, 1, 1, 1, 2, 2, 3, 3, 4, 5, 6) - output </font></td></tr>

<tr><td></td><td>  <font color="red">(3, 1, 1, 3)      </font></td><td><font color="red">  (0, 1, 1, 2, 3, 3, 4, 6, 5, 7, 8)  - remove</font></td></tr>

<tr><td>&nbsp;</td></tr>

<tr><td>5</td><td>  Empty list.</td><td></td></tr>

</tbody>
</table>
</center>
			</div>
		</div>
	</div>
</div>

<p><br></p>

<div class="faq_container">
	<div class="faq">
		<div class="faq_question">I've noticed a discrepancy between the mass of an amino acid cited in the book and in other sources.  Why is this?</div>
		<div class="faq_answer_container">
			<div class="faq_answer">
				<p>For example book suggests that glycine has elemental composition C<sub>2</sub>H<sub>3</sub>ON (integer mass 24+3+16+14=57 Da), whereas <a href="https://en.wikipedia.org/wiki/Glycine" target="_blank" rel="noopener nofollow" title="Link: https://en.wikipedia.org/wiki/Glycine">Wikipedia</a> suggests that it is C<sub>2</sub>H<sub>5</sub>ON<sub>2</sub>
  (integer mass 24+5+16+28=75 Da). We should use the former formula in
analyzing mass spectra, since when an amino acid forms a peptide bond,
it loses a water molecule (H<sub>2</sub>O).</p>
			</div>
		</div>
	</div>
</div>

<p><br></p>



<h3><a name="week4"></a>(Coursera Week 4)</h3>

<div class="faq_container">
	<div class="faq">
		<div class="faq_question">How can I improve the performance of <b>LeaderboardCyclopeptideSequencing</b>?</div>
		<div class="faq_answer_container">
			<div class="faq_answer">
				<p>You should not need to optimize your implementation for <strong>LeaderboardCyclopeptideSequencing</strong> in
 order to pass its Code Challenge.  However, various optimization
approaches can be applied.  To take one example, if the leaderboard has a
 peptide with mass smaller than <em>ParentMass</em>(<em>Spectrum</em>) but exceeding <em>ParentMass</em>(<em>Spectrum</em>)
 - 57 (recall that 57 is the mass of the lightest amino acid, glycine),
this peptide can be safely removed from the leaderboard.</p>
			</div>
		</div>
	</div>
</div>

<p><br></p>

<div class="faq_container">
	<div class="faq">
		<div class="faq_question">How can I trim the peptide leaderboard without sorting?</div>
		<div class="faq_answer_container">
			<div class="faq_answer">
				<p>To trim a peptide leaderboard without using sorting, we will first compute an array <em>ScoreHistogram,</em> where <em>ScoreHistogram</em>(<em>i</em>) holds the number of peptides in <em>Leaderboard </em>with score <em>i</em>.  For example, if we are trimming the leaderboard from Charging Station: Trimming the Peptide Leaderboard to <em>N</em> = 5 peptides (including ties), then <em>ScoreHistogram</em> =  <em>ScoreHistogram = </em>(<font color="red">0</font>, <font color="red">0</font>, <font color="red">2</font>, <font color="red">1</font>, <font color="blue">3</font>, <font color="blue">2</font>, <font color="blue">2</font>).  As a result, <font color="blue">2</font> + <font color="blue">2</font> + <font color="blue">3</font> = <font color="blue">7</font> peptides will be retained and the remaining <font color="red">0</font> + <font color="red">0</font> + <font color="red">2</font> + <font color="red">1</font> = <font color="red">3</font> peptides will be trimmed.  Here, the minimum score that a peptide can have without being cut is denoted <em>ScoreThreshold</em><sub><em>N</em></sub>(<em>Spectrum</em>).</p>

				<p><br></p>

				<p>Assuming that <em>N</em> is smaller than the number of elements on <em>Leaderboard</em>, note that the number of peptides cut is at most |<em>Leaderboard</em>| - <em>N</em>.  In order to compute <em>ScoreThreshold</em><sub><em>N</em></sub>(<em>Spectrum</em>), we need to find the index <em>i</em> such that the sum of the first <em>i</em> elements in <em>ScoreHistogram</em> is at most |<em>Leaderboard</em>| - <em>N</em> and the sum of the first <em>i</em> + 1 elements in <em>ScoreHistogram</em> exceeds |<em>Leaderboard</em>| - <em>N</em>.  To find this index, we will compute <em>CumulativeHistogram</em>, where <em>CumulativeHistogram</em>(<em>i</em>) holds the number of peptides in <em>Leaderboard</em> with score below <em>i</em>. For our ongoing example, <em>CumulativeHistogram</em> = (<font color="red">0</font>, <font color="red">0</font>, <font color="red">2</font>, <font color="red">3</font>, <font color="blue">6</font>, <font color="blue">8</font>, <font color="blue">10</font>). This leads us to the following pseudocode.</p>

				<p><br></p>

<pre><code><strong>AnotherTrim</strong>(<em>Leaderboard, Spectrum, N, AminoAcid, AminoAcidMass</em>)
&nbsp;<em></em><em>   </em><strong>for </strong><em>i</em>&nbsp;← 0 to |<em>Spectrum</em>|
<em>        ScoreHistogram</em>(<em>i</em>) ← 0
    <strong>for</strong> <em>j</em> ← 1 to |<em>Leaderboard</em>|
        <em>Peptide&nbsp;</em>←&nbsp;<em>j</em>-th peptide in <em>Leaderboard</em>
        <em>LinearSpectrum</em>  <em></em>←&nbsp;<strong>LinearSpectrum</strong>(<em>Peptide, AminoAcid, AminoAcidMass</em>)
        <em>LinearScore</em>&nbsp;←&nbsp;<em>Score</em>(<em>Peptide, </em><em>LinearSpectrum</em>)
        <em>LinearScore</em>(<em>j</em>)  <em></em>←&nbsp;<em>LinearScore</em>
        <em>ScoreHistogram</em>(<em>LinearScore</em>)&nbsp;←&nbsp;<em>ScoreHistogram</em>(<em>LinearScore</em>) + 1
        <em>sum&nbsp;</em><span style="color: rgb(51, 51, 51); font-family: 'Open Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif; line-height: 20px;">← 0
</span>        <strong>for</strong> <em>i&nbsp;</em><span style="color: rgb(51, 51, 51); font-family: 'Open Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif; line-height: 20px;">← 0</span><em>&nbsp;</em>to <em></em>|<em>Spectrum</em>|
            <em>sum&nbsp;</em>←  <em>sum</em> + <em>ScoreHistogram</em>(<em>i</em>)
        <strong>    if<em> </em></strong><em>sum</em> &gt; |<em>Leaderboard</em>| - <em>N
                ScoreThreshold&nbsp;</em>←  <em>i</em> - 1
            <strong>    for<em> </em><em>j</em>&nbsp;</strong>←<strong>&nbsp;&nbsp;</strong>1 to |<em>Leaderboard</em>|
                    <em>Peptide</em>&nbsp;←&nbsp;&nbsp;<em>j</em>-th peptide in <em></em><em>Leaderboard</em>
                <strong>    if</strong> <em>LinearScores</em>(<em><u>j</u></em>) &lt; <em>ScoreThreshold</em>
                        remove <em>Peptide</em> from <em>Leaderboard</em>
    <strong>return <em></em></strong><em></em><em>Leaderboard</em></code></pre>
			</div>
		</div>
	</div>
</div>


		</div>
	</div>
  <div id="footer" class="container_24 clearfix">
          <div class="footer-links">
     <img style="float: left;" src="/../images/alp_logo.png" alt="Active Learning Publishers" height="50" />  <a style="float: right;" href="http://rosalind.info/problems/list-view/?location=bioinformatics-textbook-track" target="_blank"><img style="padding:10px;"  src="/images/rosalind_logo_white.png" alt="Rosalind" height="50" /></a>
     &#169; 2018 by Phillip Compeau &amp; Pavel Pevzner | All Rights Reserved<br>
      ISBN: 978-0-9903746-3-3<br>
    </div>
  </div>
</div>


</body>
</html>
