<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>

<!-- Mobile Redirect -->
<script src="http://static.dudamobile.com/DM_redirect.js" type="text/javascript"></script>
<script type="text/javascript">DM_redirect("http://m.bioinformaticsalgorithms.com");</script>

<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<title>FAQs | Chapter 11 | Was T. rex Really a Big Chicken? </title>
<link rel="stylesheet" type="/text/css" href="../Styles/960_24_col.css" />
<link rel="stylesheet" type="/text/css" href="../Styles/reset.css" />
<link rel="stylesheet" type="/text/css" href="../Styles/text.css" />


<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-51449831-1', 'bioinformaticsalgorithms.com');
  ga('send', 'pageview');

</script>

<!-- Needed for JQuery in order to have expandable FAQs -->
<script type="text/javascript" src="//code.jquery.com/jquery-1.11.0.min.js"></script>

<!-- JQuery Javascript code -->
<script>
 $(document).ready(function() {

	$('.faq_question').click(function() {

		if ($(this).parent().is('.open')){
			$(this).closest('.faq').find('.faq_answer_container').animate({'height':'0'},500);
			$(this).closest('.faq').removeClass('open');

			}else{
				var newHeight =$(this).closest('.faq').find('.faq_answer').height() +'px';
				$(this).closest('.faq').find('.faq_answer_container').animate({'height':newHeight},500);
				$(this).closest('.faq').addClass('open');
			}

	});

});
</script>

</head>

<body>

    <div id="content" class="container_24 clearfix">
      <div class="main-content">
        <div class="header-img">
          <img src="../images/header-img.jpg" alt="Bioinformatics Algorithms: An Active Learning Approach" width="960" height="330" />
        </div>
        <div class="nav">
          <ul>
            <li><a href="../index.htm">Home</a></li>
            <li><a href="../about-the-author.htm">Authors</a></li>
            <li><a href="../contents.htm">Contents</a></li>
            <li><a href="../faqs.htm">FAQs</a></li>
            <li><a href="../videos.htm">Videos & Slides</a></li>
                        <li><a href="https://secure.mybookorders.com/Orderpage/1402">Buy the Book!</a></li>
            <li><a class="active" href="../contact.htm">Contact</a></li>
          </ul>
        </div>


    	<div class="section-about">

<h1>Chapter 11: Was <i>T. rex</i> Really a Big Chicken?</h1>


<div class="faq_container">
	<div class="faq">
		<div class="faq_question">Why do we include only masses of prefix and suffix subpeptides in the ideal spectrum of a peptide (e.g., <tt>RED</tt> or <tt>DCA</tt> for <tt>REDCA</tt>)? Why don't we include masses of other subpeptides (like <tt>EDC</tt>)? </div>
		<div class="faq_answer_container">
			<div class="faq_answer">
				<p>Internal subpeptides like <tt>EDC</tt> of <tt>REDCA</tt> require two bonds (before <tt>E</tt> and after <tt>C</tt>) to be broken to generate the fragment ion. In contrast, prefix and suffix subpeptides require only one bond to be broken. As a result, although internal subpeptides correspond to some fragment ions, we ignore them because they are much less common than the fragment ions generated by prefix and suffix subpeptides.</p>
			</div>
		</div>
	</div>
</div>

<p><br></p>


<div class="faq_container">
	<div class="faq">
		<div class="faq_question">How do we infer the charges of annotated peaks in a spectrum?
		</div>
		<div class="faq_answer_container">
			<div class="faq_answer"><p>For example, how did we annotate the tall peak y12++ as having charge +2 in the figure below (one of the annotations of <i>DinosaurSpectrum</i>)?</p>

		<p><br></p>

				<figure><center><img src="http://bioinformaticsalgorithms.com/images/Proteomics/dinosaur_spectrum_annotated_asara-peptide.png" width="70%"></center></figure>

				<p>As described in the main text, mass spectrometers measure the mass-to-charge ratio rather than the mass of fragment ions. Thus, a peak in a spectrum with a given mass-to-charge ratio <i>m</i>/<i>z</i> gives rise to various masses depending on its (unknown) charge <i>z</i>.  If one of the resulting masses matches a mass in the theoretical spectrum, we may infer that the peak has charge <i>z</i>.</p>
			</div>
		</div>
	</div>
</div>

<p><br></p>

<div class="faq_container">
	<div class="faq">
		<div class="faq_question">How does the use of rounded (integer) masses of amino acids diminish our ability to accurately sequence peptides?</div>
		<div class="faq_answer_container">
			<div class="faq_answer">
				<p>Although we rounded amino acid masses to integers to simplify the presentation, proteomics researchers do not round masses when attempting to interpret spectra. For example, amino acids <tt>K</tt> and <tt>Q</tt> have different molecular composition (C<sub>6</sub>H<sub>12</sub>ON<sub>2</sub> versus C<sub>5</sub>H<sub>8</sub>O<sub>2</sub>N<sub>2</sub>), and they have different monoisotopic masses (128.09497 Da versus 128.05858 Da), but their integer masses are the same. Since modern mass spectrometers are very accurate, they can detect differences in masses of up to 0.01 Da in order to distinguish <tt>K</tt> and <tt>Q</tt>.</p>
			</div>
		</div>
	</div>
</div>

<p><br></p>


<div class="faq_container">
	<div class="faq">
		<div class="faq_question">Why do the heights of some peaks in <i>DinosaurSpectrum</i> correlate so poorly with the corresponding amplitudes in the spectral vector?
		</div>
		<div class="faq_answer_container">
			<div class="faq_answer">
				<p><i>DinosaurSpectrum</i> is reproduced below (top), along with its spectral vector (bottom). </p>

				<p><br></p>

				<figure><center><img src="http://bioinformaticsalgorithms.com/images/Proteomics/dinosaur_spectrum_unannotated.png" width="70%"></center></figure>

				<figure><center><img src="http://bioinformaticsalgorithms.com/images/Proteomics/dinosaur_spectral_vector.png" width="70%"></center></figure>

				<p>The transformation of a spectrum into a spectral vector is a complex process that takes into account many factors in addition to the heights of peaks. For details of this transformation, see <a href="http://pubs.acs.org/doi/abs/10.1021/pr8001244">Kim et al., 2008.</a>.</p>

			</div>
		</div>
	</div>
</div>

<p><br></p>


<div class="faq_container">
	<div class="faq">
		<div class="faq_question">If we know the proteome, isnâ€™t it always better to use peptide identification instead of peptide sequencing?</div>
		<div class="faq_answer_container">
			<div class="faq_answer">
				<p>If a spectrum that we analyze originated from a peptide in a proteome, then it makes sense to apply peptide identification via database search and to identify this peptide. However, our knowledge of the proteomes remains incomplete even in the case of a well-studied human proteome. Biologists therefore sometimes use <i>de novo</i> peptide sequencing to discover peptides that do not appear in the currently known (still incomplete) proteome.</p>
			</div>
		</div>
	</div>
</div>

<p><br></p>

<div class="faq_container">
	<div class="faq">
		<div class="faq_question">Why do we generate the decoy database by assuming that all amino acids have the same frequency (1/20), despite the fact that many proteomes have widely varying amino acid frequencies? </div>
		<div class="faq_answer_container">
			<div class="faq_answer">
				<p>In practice, proteomics researchers do typically generate decoy databases by taking into account amino acid frequencies in the proteome under study. This is often achieved by randomly shuffling the amino acids in real proteins to generate a decoy database.</p>
			</div>
		</div>
	</div>
</div>

<p><br></p>

<div class="faq_container">
	<div class="faq">
		<div class="faq_question">What is the running time of the dynamic programming algorithm for computing the size of a spectral dictionary?</div>
		<div class="faq_answer_container">
			<div class="faq_answer">
				<p>The algorithm for computing |<i>Dictionary</i><sub><i>threshold</i></sub>(<i>Spectrum</i>)| amounts to filling up the <i>N</i>*<i>M</i> table <i>Size</i>(<i>i</i>, <i>t</i>), where <i>N</i> is the maximum score among all peptides against <i>Spectrum</i> and <i>M</i> is the parent mass of <i>Spectrum</i>. To fill in each element of this table, we need to consider all possible amino acids and apply the equation for <i>Size</i>(<i>i</i>, <i>t</i>) from the section "Spectral Dictionaries".</p>
			</div>
		</div>
	</div>
</div>

<p><br></p>

<!--
<div class="faq_container">
	<div class="faq">
		<div class="faq_question">Question here</div>
		<div class="faq_answer_container">
			<div class="faq_answer">
				Answer here
			</div>
		</div>
	</div>
</div>

<p><br></p>
-->


		</div>
	</div>
  <div id="footer" class="container_24 clearfix">
          <div class="footer-links">
     <img style="float: left;" src="/../images/alp_logo.png" alt="Active Learning Publishers" height="50" />  <a style="float: right;" href="http://rosalind.info/problems/list-view/?location=bioinformatics-textbook-track" target="_blank"><img style="padding:10px;"  src="/images/rosalind_logo_white.png" alt="Rosalind" height="50" /></a>
     &#169; 2018 by Phillip Compeau &amp; Pavel Pevzner | All Rights Reserved<br>
      ISBN: 978-0-9903746-3-3<br>
    </div>
  </div>
</div>


</body>
</html>
