<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>

<!-- Mobile Redirect -->
<script src="http://static.dudamobile.com/DM_redirect.js" type="text/javascript"></script>
<script type="text/javascript">DM_redirect("http://m.bioinformaticsalgorithms.com");</script>

<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<title>FAQs | Chapter 6 | Are There Fragile Regions in the Human Genome? </title>
<link rel="stylesheet" type="/text/css" href="../Styles/960_24_col.css" />
<link rel="stylesheet" type="/text/css" href="../Styles/reset.css" />
<link rel="stylesheet" type="/text/css" href="../Styles/text.css" />


<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-51449831-1', 'bioinformaticsalgorithms.com');
  ga('send', 'pageview');

</script>


<!-- Needed for JQuery in order to have expandable FAQs -->
<script type="text/javascript" src="//code.jquery.com/jquery-1.11.0.min.js"></script>

<!-- JQuery Javascript code -->
<script>
 $(document).ready(function() {

	$('.faq_question').click(function() {

		if ($(this).parent().is('.open')){
			$(this).closest('.faq').find('.faq_answer_container').animate({'height':'0'},500);
			$(this).closest('.faq').removeClass('open');

			}else{
				var newHeight =$(this).closest('.faq').find('.faq_answer').height() +'px';
				$(this).closest('.faq').find('.faq_answer_container').animate({'height':newHeight},500);
				$(this).closest('.faq').addClass('open');
			}

	});

});
</script>

</head>

<body>

    <div id="content" class="container_24 clearfix">
      <div class="main-content">
        <div class="header-img">
          <img src="../images/header-img.jpg" alt="Bioinformatics Algorithms: An Active Learning Approach" width="960" height="330" />
        </div>
        <div class="nav">
          <ul>
            <li><a href="../index.htm">Home</a></li>
            <li><a href="../about-the-author.htm">Authors</a></li>
            <li><a href="../contents.htm">Contents</a></li>
            <li><a href="../faqs.htm">FAQs</a></li>
            <li><a href="../videos.htm">Videos & Slides</a></li>
                        <li><a href="https://secure.mybookorders.com/Orderpage/1402">Buy the Book!</a></li>
            <li><a class="active" href="../contact.htm">Contact</a></li>
          </ul>
        </div>


    	<div class="section-about">

<h1>Chapter 6: Are There Fragile Regions in the Human Genome?</h1>

<h3><a name="week4"></a>(Coursera Week 4)</h3>

<div class="faq_container">
	<div class="faq">
		<div class="faq_question">Since humans did not descend directly from mice, why do we nevertheless analyze a rearrangement scenario transforming mouse into human?</div>
		<div class="faq_answer_container">
			<div class="faq_answer">
				<p>Indeed, mouse and human have a common ancestor from which they have both evolved. Yet when we construct a scenario consisting of <i>n</i> rearrangements transforming the mouse genome into the human genome, the first <i>x</i> rearrangements represent a transformation of the mouse genome into the ancestor genome (going back in time) and the last <i>n</i>-<i>x</i> rearrangements represent a transformation from the ancestor to the human genome. This relies on the fact that the rearrangements we consider are <i>invertible</i>, e.g., the inverse operation of a reversal is a reversal.</p>
			</div>
		</div>
	</div>
</div>

<p><br></p>

<div class="faq_container">
	<div class="faq">
		<div class="faq_question">Why does the Random Breakage Model result in an exponential distribution of synteny blocks?</div>
		<div class="faq_answer_container">
			<div class="faq_answer">
				<p>In a <b>Poisson distribution</b>, we assume that some event is happening on average λ times within a given interval of fixed length, with no relationship between the occurrences.  That is, if we look at a given interval, we will see on average λ occurrences, but there may be any finite number of occurrences in practice.  If the Random Breakage Model is true, then the Poisson distribution offers a good model for the number of breakpoints that occur in a given interval of the genome.</p>

				<p><br></p>

				<p>Letting <i>N</i><sub><i>t</i></sub> denote the number of events occurring between point 0 and point <i>t</i>, it can be shown that the probability of exactly <i>k</i> events in an interval of unit length is given by the formula</p>

				<p><br></p>

				<p><center>Pr(<i>N</i><sub>1</sub> = <i>k</i>) = (λ<sup><i>k</i></sup><i>e</i><sup>-λ</sup>)/<i>k</i>!</center></p>

				<p><br></p>

				<p>where <i>k</i> is allowed to be any nonnegative integer and <i>e</i> = 2.7182818284... is the base of the natural logarithm.</p>

				<p><br></p>

				<p>In particular, the probability of no events (<i>N</i><sub>1</sub> = 0) is given by <i>e</i><sup>-λ</sup>.  Note that if we scale the interval's unit of measurement to any positive real number <i>x</i> (measured in units of the original interval), we obtain that Pr(<i>N</i><sub><i>x</i></sub> = 0) is instead <i>e</i><sup>-λ<i>x</i></sup>.</p>

				<p><br></p>

				<p>Now, we are interested in not the <i>number</i> of events (i.e., breakpoints) occurring in a given interval of time, but rather, the distribution of the distances between events (i.e., distances between breakpoints along the genome). Accordingly, let <i>X</i><sub><i>t</i></sub> denote the random variable corresponding to the time needed for the next event, given that the last event occurred at time <i>t</i>.  We will first write</p>

				<p><br></p>

				<p><center>Pr(<i>X</i><sub><i>t</i></sub> ≤ <i>x</i>) = 1 - Pr(<i>X</i><sub><i>t</i></sub> > <i>x</i>)</center></p>

				<p><br></p>

				<p>Note that Pr(<i>X</i><sub><i>t</i></sub> > <i>x</i>) is equal to Pr(<i>N</i><sub><i>t</i></sub> = <i>N</i><sub><i>t</i>+<i>x</i></sub>), and so</p>

				<p><br></p>

				<p><center>Pr(<i>X</i><sub><i>t</i></sub> ≤ <i>x</i>) = 1 - Pr(<i>N</i><sub><i>t</i></sub> = <i>N</i><sub><i>t</i>+<i>x</i></sub>).</center></p>

				<p><br></p>

				<p>The Poisson distribution is "memoryless", meaning that the interval between <i>t</i> and <i>t</i>+<i>x</i> is equivalent to the interval between 0 and <i>x</i> in terms of probability, so that we can conclude that</p>

				<p><br></p>

				<p><center>Pr(<i>X</i><sub><i>t</i></sub> ≤ <i>x</i>) = 1 - Pr(<i>N</i><sub><i>x</i></sub>) = 0.</center></p>

				<p><br></p>

				<p>The probability on the right is what we calculated above, and so we conclude that Pr(<i>X</i><sub><i>t</i></sub> ≤ <i>x</i>) = 1-<i>e</i><sup>-λ<i>x</i></sup>.  This is the cumulative distribution function of an exponential random variable, which is what we set out to demonstrate.</p>

				<p><br></p>

				<p>Notation adapted from <a href="http://stats.stackexchange.com/questions/2092/relationship-between-poisson-and-exponential-distribution">http://stats.stackexchange.com/questions/2092/relationship-between-poisson-and-exponential-distribution</a> </p>
			</div>
		</div>
	</div>
</div>


				<p><br></p>

<div class="faq_container">
	<div class="faq">
		<div class="faq_question">What is a permutation?</div>
		<div class="faq_answer_container">
			<div class="faq_answer">
				<p>A <b>permutation</b> is a specific ordering of the positive integers from 1 to <i>n</i>, where each element is used exactly once. For example, there are six permutations of length 3:</p>

				<p><br></p>

<p><center>(1 2 3)&nbsp;&nbsp;&nbsp;(1 3 2)&nbsp;&nbsp;&nbsp;(2 1 3)&nbsp;&nbsp;&nbsp;(2 3 1)&nbsp;&nbsp;&nbsp;(3 1 2)&nbsp;&nbsp;&nbsp;(3 2 1)</center></p>

<p><br></p>

<p>In this book, we often use the term "permutation" as shorthand for a <b>signed permutation</b>, in which each element has a sign, or orientation (represented as a "+" or "-").  You can verify that there are 48 signed permutations of length 3. </p>
			</div>
		</div>
	</div>
</div>

<p><br></p>

<div class="faq_container">
	<div class="faq">
		<div class="faq_question">How do we compare genomes where some synteny blocks appear in multiple copies, such as (+<i>a</i> +<i>b</i> +<i>c</i> +<i>b</i>) and (+<i>a</i> -<i>b</i> +<i>b</i> -<i>c</i>)?  </div>
		<div class="faq_answer_container">
			<div class="faq_answer">
				<p>You can label repeated elements in the first genome using subscripts so that each synteny block appears just once, e.g., (+<i>a</i> +<i>b</i><sub>1</sub> +<i>c</i> +<i>b</i><sub>2</sub>). You can then label the second genome either as (+<i>a</i> -<i>b</i><sub>1</sub> +<i>b</i><sub>2</sub> -<i>c</i>) or as (+<i>a</i> -<i>b</i><sub>2</sub> +<i>b</i><sub>1</sub> -<i>c</i>) and compute the 2-break distance from (+<i>a</i> +<i>b</i><sub>1</sub> +<i>c</i> +<i>b</i><sub>2</sub>) to each of the two resulting genomes, selecting the one that results in the minimum 2-break distance as the best labeling.</p>

				<p><br></p>

				<p>The problem with this approach is that the number of re-labelings of a permutation with duplicated elements may grow very quickly.  Furthermore, this approach only works when the number of copies of the same synteny block in each of genome is the same.
</p>
			</div>
		</div>
	</div>
</div>



<p><br></p>

<div class="faq_container">
	<div class="faq">
		<div class="faq_question">How do we compare genomes with different numbers of synteny blocks, such as (+<i>a</i> +<i>b</i> +<i>c</i>) and (+<i>c</i> -<i>b</i> +<i>a</i> -<i>d</i>)?  </div>
		<div class="faq_answer_container">
			<div class="faq_answer">
				<p>The easiest way to deal with synteny blocks that appear in one genome and not another is to ignore them and consider only those blocks common to both genomes, e.g., in this case to compare (+<i>a</i> +<i>b</i> +<i>c</i>) with (+<i>c</i> -<i>b</i> +<i>a</i>). It is also possible to incorporate insertions and deletions into genome rearrangement studies, providing some penalty for the insertion/deletion of a single block, or a penalty for the insertion/deletion of a series of contiguous blocks. Various research papers have attempted to expand genome rearrangement metrics to account for insertions and deletions.</p>
			</div>
		</div>
	</div>
</div>


<p><br></p>

<div class="faq_container">
	<div class="faq">
		<div class="faq_question">How can we conclude that there are 1,070 different seven-step scenarios to transform the mouse X chromosome into the human X chromosome by reversals?</div>
		<div class="faq_answer_container">
			<div class="faq_answer">
				<p>Given a permutation <i>P</i> and a reversal ρ, we denote the genome resulting from applying ρ to <i>P</i> as <i>P</i>*ρ. A reversal ρ is called <b><i>P</i>-valid</b> if the reversal distance of <i>P</i>*ρ is smaller than the 2-break distance of <i>P</i>. The following recurrence relation computes <i>NumberOfScenarios</i>(<i>P</i>), the number of different reversal scenarios that transform a genome <i>P</i> into the identity permutation using the minimum number of reversals:</p>

<p><br></p>

<p><center><i>NumberOfScenarios</i>(<i>P</i>) = Σ<sub>all <i>P</i>-valid 2-breaks ρ</sub>NumberOfScenarios(<i>P</i>*ρ)</center></p>
			</div>
		</div>
	</div>
</div>

<p><br></p>

<div class="faq_container">
	<div class="faq">
		<div class="faq_question">Why does the pair (+4 +3) form a breakpoint but the pair (-4 -3) does not?</div>
		<div class="faq_answer_container">
			<div class="faq_answer">
				<p>The pair (<font color="blue">+4 +3</font>) forms a breakpoint because, in contrast to (<font color="green">-4 -3</font>), it cannot be transformed into (+3 +4), a desirable pair when sorting by reversals, by a single reversal. For example,  applying a reversal to</p>

<p><br></p>

<p><center>(+1 +2 <font color="blue">+4 +3</font> +5 +6)</center></p>

<p><br></p>

<p>transforms this permutation into</p>

<p><br></p>

<p><center>(+1 +2 <font color="blue">-3 -4</font> +5 +6),</center></p>

<p><br></p>

<p>but applying a reversal to</p>

<p><br></p>

<p><center>(+1 +2 <font color="green">-4 -3</font> +5 +6)</center></p>

<p><br></p>

<p>transforms it into the identity permutation</p>

<p><br></p>

<p><center>(+1 +2 <font color="green">+3 +4</font> +5 +6).</center></p>

<p><br></p>

<p>To better understand why (<font color="blue">+4 +3</font>) is a breakpoint, try sorting the permutation (+6 +5 +4 +3 +2 +1) – you will see that it requires many reversals!</p>

			</div>
		</div>
	</div>
</div>

<p><br></p>


<div class="faq_container">
	<div class="faq">
		<div class="faq_question">Are there other types of genome rearrangements other than reversals, translocations, fusions, and fissions?</div>
		<div class="faq_answer_container">
			<div class="faq_answer">
				<p>Yes! For example, a transposition moves a segment from one location in the genome to another. For example, one transposition applied to the blue region of the chromosome (+1 <font color="blue">+2 +3 +4</font> <font color="red">+5 +6</font> +7) yields (+1 <font color="red">+5 +6</font> <font color="blue">+2 +3 +4</font> +7). However, transpositions are more rare than reversals and other rearrangements discussed in the chapter.</p>

                <p><br></p>

                <p>Transpositions represent an example of a <b>3-break</b>, a rearrangement that requires 3 rather than 2 breaks (between +1 and +2, between +4 +5, and between +6 and +7). Since 3-breaks are rare compared to 2-breaks, we can obtain reasonable distance functions without them, and so 3-breaks are not covered in this chapter.</p>
			</div>
		</div>
	</div>
</div>

<p><br></p>



<h3><a name="week5"></a>(Coursera Week 5)</h3>



<div class="faq_container">
	<div class="faq">
		<div class="faq_question">The Exercise Break in the section “Breakpoint Graphs” suggests that the only genome that forms a trivial breakpoint graph with a genome <i>P</i> is <i>P</i> itself. But I can find another genome satisfying this condition!</div>
		<div class="faq_answer_container">
			<div class="faq_answer">
				<p>The section “Breakpoint Graphs” shows a trivial breakpoint graph <i>BreakpointGraph</i>(<i>P</i>, <i>P</i>) for <i>P</i> = (+<i>a</i> –<i>b</i> –<i>c</i> +<i>d</i>). Another trivial breakpoint graph is seemingly formed by the genomes <i>P</i> and <i>Q</i> = (-<i>a</i> +<i>b</i> +<i>c</i> -<i>d</i>). But note that <i>P</i> and <i>Q</i> represent the <i>same</i> circular chromosome traversed in opposite directions; therefore, <i>P</i> and <i>Q</i> are indeed identical.  </p>
			</div>
		</div>
	</div>
</div>

<p><br></p>


<div class="faq_container">
	<div class="faq">
		<div class="faq_question">Can you give an example of how <b>GraphToGenome</b> should work?</div>
		<div class="faq_answer_container">
			<div class="faq_answer">
				<p>The following explanation is a modification of one given by one of our excellent community TAs, Giampaolo Eusebi.</p>

				<p><br></p>

				<p>Keep in mind that:</p>

				<ul>
				<li>Every even number 2<i>x</i> is a black node head, and every even number 2<i>x</i>−1 is a black node tail;</li>
				<li>every colored edge is composed by two numbers representing black heads or tails.</li>
				</ul>

				<p>That being said, the order should not be very important. Take, for example, the following edge list:</p>

				<p><br></p>

				<p><center>(2,4), (7,9), (10,12), (3,6), (5,1), (11,8)</center></p>

				<p><br></p>

				<p>If you start with (2,4):</p>

				<ul>
				<li>(2,4) ends with a 4 (even), and the only edge that starts with 4−1=3 is (3,6);</li>
				<li>(3,6) ends with a 6 (even), and the only edge that starts with 6−1=5 is (5,1);</li>
				<li>(5,1) ends with a 1 (odd), and the only edge that starts with 1+1=2 is (2,4), which brings us back where we started, thus forming a cycle.</li>
				</ul>

				<p>The only remaining edges are (7,9), (10,12),(11,8). If you start with (7,9):</p>

				<ul>
				<li>(7,9) ends with a 9 (odd), and the only edge that starts with 9+1=10 is (10,12);</li>

				<li>(10,12) ends with a 12 (even), and the only edge that starts with 12−1=11 is (11,8);</li>

				<li>(11,8) ends with a 8 (even), and the only edge that starts with 8−1=7 is (7,9), which brings us back where we started, thus forming a cycle.</li>
				</ul>

				<p>The edge list is now empty.  The key point is that we will have obtained the same two cycles regardless of which edges we chose as starting points (feel free to try it for yourself).</p>
			</div>
		</div>
	</div>
</div>

<p><br></p>

<div class="faq_container">
	<div class="faq">
		<div class="faq_question">We refuted the Random Breakage Model by assuming that human and mouse genomes have circular chromosomes. But don't these genomes have linear chromosomes?</div>
		<div class="faq_answer_container">
			<div class="faq_answer">
				<p>We used 2-break distance for circular chromosomes to refute the Random Breakage Model. See “DETOUR: Sorting linear permutations by reversals” or Bergeron, Mixtacki, Stoye 2006 (<a href="https://pub.uni-bielefeld.de/publication/1596811">https://pub.uni-bielefeld.de/publication/1596811</a>) to see that similar formulas hold for linear chromosomes. </p>
			</div>
		</div>
	</div>
</div>

<p><br></p>

<div class="faq_container">
	<div class="faq">
		<div class="faq_question">Why do we ignore small diagonals when constructing synteny blocks?</div>
		<div class="faq_answer_container">
			<div class="faq_answer">
				<p>In addition to the dots representing conserved genes between two species, genomic dot-plots contain many spurious dots. As discussed in the main text, even randomly generated strings have shared <i>k</i>-mers that result in "spurious" dots in their genomic dot-plots. Moreover, these spurious <i>k</i>-mers may aggregate into spurious diagonals that must be removed for follow-up analysis of synteny blocks. Since these spurious diagonals are usually short, we filter out short diagonals when constructing synteny blocks.</p>
			</div>
		</div>
	</div>
</div>


<p><br></p>

<div class="faq_container">
	<div class="faq">
		<div class="faq_question">How can we account for mutations when constructing synteny blocks? </div>
		<div class="faq_answer_container">
			<div class="faq_answer">
				<p>Our algorithm for constructing synteny blocks, which is based on shared k-mers, does account for mutations. For example, even though the two "genes" <tt>ACTG<font color="red">A</font>GTTC</tt> and <tt>ACTG<font color="red">G</font>GTTC</tt> differ from each other by a mutation (<tt>A</tt> -> <tt>G</tt>), the genomic dot-plot with <i>k</i> = 3 will reveal that they form a single synteny block; construct this dot-plot and see for yourself!</p>

                <p><br></p>

                <p>Modern programs for constructing synteny blocks use dot-pots representing all local alignments (with scores exceeding a threshold) rather than  all shared <i>k</i>-mers  between the two genomes. However, constructing all such local alignments for long genomes is a time-consuming task.</p>
			</div>
		</div>
	</div>
</div>



<p><br></p>

<div class="faq_container">
	<div class="faq">
		<div class="faq_question">How do reverse palindromes affect the construction of genomic dot-plots?</div>
		<div class="faq_answer_container">
			<div class="faq_answer">
				<p>As specified in the main text:</p>

<blockquote>We color the point (<em>x</em>, <em>y</em>) red if the two genomes share a <em>k</em>-mer at respective positions <em>x</em> and <em>y</em>. We color (<em>x</em>, <em>y</em>) blue if the <em>k</em>-mer starting at position <em>x</em> in the first genome is the reverse complement of the <em>k</em>-mer starting at position <em>y</em> in the second genome.</blockquote>

<p>This definition does not specify what to do with reverse palindromes. A <b>reverse palindrome</b> is a DNA string that is its own reverse complement, such as <tt>ACGCGT</tt>. If a reverse complement starts at respective positions <em>x</em> and <em>y</em> in two genomes, then (<em>x</em>, <em>y</em>) should technically be colored both red and blue! To address this issue, we will use only red to color points corresponding to reverse palindromes.</p>
			</div>
		</div>
	</div>
</div>

<p><br></p>

<div class="faq_container">
	<div class="faq">
		<div class="faq_question">Can we use sorting to solve the Shared <em>k</em>-mers Problem?</div>
		<div class="faq_answer_container">
			<div class="faq_answer">
				<p>Yes. For example, the strings <tt><strong>AGCAGGTTATCTACCTGT</strong></span></tt> and <strong><tt><font color="red">AGCAGGAGATAAACCTGT</font></tt></strong> can be transformed into sequences of their 3-mers along with these 3-mers' respective starting positions:</p>

				<p><br></p>

				<p><tt>(<strong>AGC</strong>,0) (<strong>GCA</strong>,1) (<strong>CAG</strong>,2) (<strong>AGG</strong>,3) (<strong>GGT</strong>,4) (<strong>GTT</strong>,5) (<strong>TTA</strong>,6) (<strong>TAT</strong>,7) (<strong>ATC</strong>,8) (<strong>TCT</strong>,9) (<strong>CTA</strong>,10) (<strong>TAC</strong>,11), (<strong>ACC</strong>,12) (<strong>CCT</strong>,13) (<strong>CTG</strong>,14) (<strong>TGT</strong>,15)</tt></p>

				<p><br></p>

				<p><tt>(<font color="red"><strong>AGC</strong></font>,0) (<font color="red"><strong>GCA</strong></font>,1) (<font color="red"><strong>CAG</strong></font>,2) (<font color="red"><strong>AGG</strong></font>,3) (<font color="red"><strong>GGA</strong></font>,4) (<font color="red"><strong>GAG</strong></font>,5) (<font color="red"><strong>AGA</strong></font>,6) (<font color="red"><strong>GAT</strong></font>,7) (<font color="red"><strong>ATA</strong></font>,8) (<font color="red"><strong>TAA</strong></font>,9) (<font color="red"><strong>AAA</strong></font>,10) (<font color="red"><strong>AAC</strong></font>,11) (<font color="red"><strong>ACC</strong></font>,12) (<font color="red"><strong>CCT</strong></font>,13) (<font color="red"><strong>CTG</strong></font>,14) (<font color="red"><strong>TGT</strong></font>,15)</tt></p>

				<p><br></p>

				<p>We now take the reverse complement of each 3-mer of the second string <tt><font color="red"><strong>AGCAGGACATAAACCTGT</strong></font></tt> and color them blue:</p>

				<p><br></p>

				<p><tt>(<font color="blue"><strong>GCT</strong></font>,0) (<font color="blue"><strong>TGC</strong></font>,1) (<font color="blue"><strong>CTG</strong></font>,2) (<font color="blue"><strong>CCT</strong></font>,3) (<font color="blue"><strong>TCC</strong></font>,4) (<font color="blue"><strong>CTC</strong></font>,5) (<font color="blue"><strong>TCT</strong></font>,6) (<font color="blue"><strong>ATC</strong></font>,7) (<font color="blue"><strong>TAT</strong></font>,8) (<font color="blue"><strong>TTA</strong></font>,9) (<font color="blue"><strong>TTT</strong></font>,10) (<font color="blue"><strong>GTT</strong></font>,11) (<font color="blue"><strong>GGT</strong></font>,12) (<font color="blue"><strong>AGG</strong></font>,13) (<font color="blue"><strong>CAG</strong></font>,14) (<font color="blue"><strong>ACA</strong></font>,15)</tt></p>

				<p><br></p>

				<p>Afterwards, we lexicographically merge and sort all three arrays into a single array:</p>

				<p><br></p>

				<p><tt>(<font color="red"><strong>AAA</strong></font>,10) (<font color="red"><strong>AAC</strong></font>,11) (<font color="blue"><strong>ACA</strong></font>,15) (<strong>ACC</strong>,12)  (<font color="red"><strong>ACC</strong></font>,12) (<font color="red"><strong>AGA</strong></font>,6) (<strong>AGC</strong>,0) (<font color="red"><strong>AGC</strong></font>,0) (<strong>AGG</strong>,3) (<font color="red"><strong>AGG</strong></font>,3) (<font color="blue"><strong>AGG</strong></font>,13) (<font color="blue"><strong>GTT</strong></font>,11) (<font color="red"><strong>ATA</strong></font>,8) (<strong>ATC</strong>,8) (<font color="blue"><strong>ATC</strong></font>,7) (<font color="red"><strong>CAG</strong></font>,2) (<strong>CAG</strong>,2) (<font color="blue"><strong>CAG</strong></font>,14) (<font color="red"><strong>CCT</strong></font>,13)  (<strong>CCT</strong>,13) (<font color="blue"><strong>CCT</strong></font>,3) (<strong>CTA</strong>,10) (<font color="blue"><strong>CTC</strong></font>,5) (<strong>CTG</strong>,14) (<font color="blue"><strong>CTG</strong></font>,2) (<font color="red"><strong>CTG</strong></font>,14) (<font color="red"><strong>GAG</strong></font>,5)  (<font color="red"><strong>GAT</strong></font>,7) (<strong>GCA</strong>,1) (<font color="red"><strong>GCA</strong></font>,1) (<font color="blue"><strong>GCT</strong></font>,0) (<font color="red"><strong>GGA</strong></font>,4) (<strong>GGT</strong>,4) (<font color="blue"><strong>GGT</strong></font>,12) (<font color="red"><strong>TAA</strong></font>,9) (<strong>TAC</strong>,11) (<strong>TAT</strong>,7) (<font color="blue"><strong>TAT</strong></font>,8)  (<font color="blue"><strong>TCC</strong></font>,4) (<strong>TCT</strong>,9) (<font color="blue"><strong>TCT</strong></font>,6) (<font color="blue"><strong>TGC</strong></font>,1) (<strong>TGT</strong>,15) (<font color="red"><strong>TGT</strong></font>,15) (<strong>TTA</strong>,6)  (<font color="blue"><strong>TTA</strong></font>,9) (<font color="blue"><strong>TTT</strong></font>,10)</tt></p>

				<p><br></p>

				<p>We will retain only 3-mers that appear as a black 3-mer and a colored 3-mer:</p>

				<p><br></p>

				<p><tt>(<strong>ACC</strong>,12) (<font color="red"><strong>ACC</strong></font>,12) (<strong>AGC</strong>,0) (<font color="red"><strong>AGC</strong></font>,0) (<strong>AGG</strong>,3) (<font color="red"><strong>AGG</strong></font>,3) (<font color="blue"><strong>AGG</strong></font>,13) (<strong>ATC</strong>,8) (<font color="blue"><strong>ATC</strong></font>,7) (<strong>CAG</strong>,2) (<font color="red"><strong>CAG</strong></font>,2) (<font color="blue"><strong>CAG</strong></font>,14) (</font><strong>CCT</strong>,13) </font>(</font><font color="red"><strong>CCT</strong></font>,13) (<font color="blue"><strong>CCT</strong></font>,3) (<strong>CTG</strong>,14) (<font color="blue"><strong>CTG</strong></font>,2) (<font color="red"><strong>CTG</strong></font>,14) (<strong>GCA</strong>,1) (<font color="red"><strong>GCA</strong></font>,1) (<strong>GGT</strong>,4) (<font color="blue"><strong>GGT</strong></font>,12) (<strong>GTT</strong>,5) (<font color="blue"><strong>GTT</strong></font>,11) (<strong>TAT</strong>,7) (<font color="blue"><strong>TAT</strong></font>,8)  (<strong>TCT</strong>,9) (<font color="blue"><strong>TCT</strong></font>,6) (<strong>TGT</strong>,15) (<font color="red"><strong>TGT</strong></font>,15) (<strong>TTA</strong>,6)  (<font color="blue"><strong>TTA</strong></font>,9)</tt></p>

				<p><br></p>

				<p>If a 3-mer appears in this list as black and red with positions <em>x</em> and <em>y</em>, then we add a red point (<em>x, </em><em>y</em>) to the genomic dot-plot (note: in this case <em>x</em> = <em>y</em>).  If a 3-mer appears in this list as black and blue with respective positions <em>x</em> and <em>y</em>, then we add a blue point (<em>x, </em><em>y</em>) to the genomic dot-plot. See the figure below.</p>

				<p><br></p>

				<p><tt>(<strong>ACC</strong>,12) (<font color="red"><strong>ACC</strong></font>,12)               (<font color="red"><strong>12</strong></font>,<font color="red"><strong>12</strong></font>)<br>
				(<strong>AGC</strong>,0)  (<font color="red"><strong>AGC</strong></font>,0)                (<font color="red"><strong>0</strong></font>,<font color="red"><strong>0</strong></font>)<br>
				</font>(<strong>AGG</strong>,3)  (<font color="red"><strong>AGG</strong></font>,3)  (<font color="blue"><strong>AGG</strong></font>,13)      (<font color="red"><strong>3</strong></font>,<font color="red"><strong>3</strong></font>), (<font color="blue"><strong>3</strong></font>,<font color="blue"><strong>13</strong></font>) <br>
</font>(<strong>ATC</strong>,8)  (<font color="blue"><strong>ATC</strong></font>,7)                (<font color="blue"><strong>8</strong></font>,<font color="blue"><strong>7</strong></font>)<br>
</font>(<strong>CAG</strong>,2)  (<font color="red"><strong>CAG</strong></font>,2)  (<font color="blue"><strong>CAG</strong></font>,14)      (<font color="red"><strong>2</strong></font>,<font color="red"><strong>2</strong></font>), (<font color="blue"><strong>2</strong></font>,<font color="blue"><strong>14</strong></font>)<br>
</font>(<strong>CCT</strong>,13) (<font color="red"><strong>CCT</strong></font>,13) (<font color="blue"><strong>CCT</strong></font>,3)       (<font color="red"><strong>13</strong></font>,<font color="red"><strong>13</strong></font>), (<font color="blue"><strong>13</strong></font>,<font color="blue"><strong>3</strong></font>)<br>
</font>(<strong>CTG</strong>,14) (<font color="blue"><strong>CTG</strong></font>,2)  (<font color="red"><strong>CTG</strong></font>,14)      (<font color="blue"><strong>14</strong></font>,<font color="blue"><strong>2</strong></font>), (<font color="red"><strong>14</strong></font>,<font color="red"><strong>14</strong></font>)<br>
</font>(<strong>GCA</strong>,1)  (<font color="red"><strong>GCA</strong></font>,1)                (<font color="red"><strong>1</strong></font>,<font color="red"><strong>1</strong></font>)<br>
</font>(<strong>GGT</strong>,4)  (<font color="blue"><strong>GGT</strong></font>,12)               (<font color="blue"><strong>4</strong></font>,<font color="blue"><strong>12</strong></font>)<br>
</font>(</font><strong>GTT</strong>,5)  </font>(</font><font color="blue"><strong>GTT</strong></font>,11)               </font>(<font color="#0000ff"><b>5</b></font></font>,</font><font color="blue"><strong>11</strong></font>)</font><br>
</font>(</font><strong>TAT</strong>,7)  (</font><font color="blue"><strong>TAT</strong></font>,8)                (</font><font color="blue"><strong>7</strong></font>,</font><font color="blue"><strong>8</strong></font>)<br>
</font>(</font><strong>TCT</strong>,9)  (</font><font color="blue"><strong>TCT</strong></font>,6)                (</font><font color="blue"><strong>9</strong></font>,</font><font color="blue"><strong>6</strong></font>)<br>
</font>(</font><strong>TGT</strong>,15) (</font><font color="red"><strong>TGT</strong></font>,15)               (</font><font color="red"><strong>15</strong></font>,</font><font color="red"><strong>15</strong></font>)<br>
</font>(</font><strong>TTA</strong>,6)  (</font><font color="blue"><strong>TTA</strong></font>,9)                (</font><font color="blue"><strong>6</strong></font>,</font><font color="blue"><strong>9</strong></font>)</tt></p>

				<p><br></p>

				<figure><center><img src="../images/Rearrangements/faq_dot_plot.png" width="55%"></center></figure>

			</div>
		</div>
	</div>
</div>


<p><br></p>

<div class="faq_container">
	<div class="faq">
		<div class="faq_question">In "DETOUR: Sorting Linear Permutations by Reversals", why do we need the complex formula for reversal distance if we have a simpler formula for the 2-break distance for linear chromosomes?</div>
		<div class="faq_answer_container">
			<div class="faq_answer">
				<p>2-breaks include reversals, but not every 2-break is a reversal. For example, one 2-break on the linear chromosome (+<i>a</i> +<i>b</i> <font color="red">+<i>c</i> +<i>d</i></font> +<i>e</i>) may yield a fission operation, resulting in the linear chromosome (+<i>a</i> +<i>b</i> +<i>e</i>) and the circular chromosome (<font color="red">+<i>c</i> +<i>d</i></font>).</p>
			</div>
		</div>
	</div>
</div>





<!--
<div class="faq_container">
	<div class="faq">
		<div class="faq_question">Question here</div>
		<div class="faq_answer_container">
			<div class="faq_answer">
				Answer here
			</div>
		</div>
	</div>
</div>
-->



		</div>
	</div>
  <div id="footer" class="container_24 clearfix">
          <div class="footer-links">
     <img style="float: left;" src="/../images/alp_logo.png" alt="Active Learning Publishers" height="50" />  <a style="float: right;" href="http://rosalind.info/problems/list-view/?location=bioinformatics-textbook-track" target="_blank"><img style="padding:10px;"  src="/images/rosalind_logo_white.png" alt="Rosalind" height="50" /></a>
     &#169; 2018 by Phillip Compeau &amp; Pavel Pevzner | All Rights Reserved<br>
      ISBN: 978-0-9903746-3-3<br>
    </div>
  </div>
</div>


</body>
</html>
