<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>

<!-- Mobile Redirect -->
<script src="http://static.dudamobile.com/DM_redirect.js" type="text/javascript"></script>
<script type="text/javascript">DM_redirect("http://m.bioinformaticsalgorithms.com");</script>

<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<title>FAQs | Application Challenge | RNA-Seq </title>
<link rel="stylesheet" type="/text/css" href="../Styles/960_24_col.css" />
<link rel="stylesheet" type="/text/css" href="../Styles/reset.css" />
<link rel="stylesheet" type="/text/css" href="../Styles/text.css" />


<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-51449831-1', 'bioinformaticsalgorithms.com');
  ga('send', 'pageview');

</script>


<!-- Needed for JQuery in order to have expandable FAQs -->
<script type="text/javascript" src="//code.jquery.com/jquery-1.11.0.min.js"></script>

<!-- JQuery Javascript code -->
<script>
 $(document).ready(function() {

	$('.faq_question').click(function() {

		if ($(this).parent().is('.open')){
			$(this).closest('.faq').find('.faq_answer_container').animate({'height':'0'},500);
			$(this).closest('.faq').removeClass('open');

			}else{
				var newHeight =$(this).closest('.faq').find('.faq_answer').height() +'px';
				$(this).closest('.faq').find('.faq_answer_container').animate({'height':newHeight},500);
				$(this).closest('.faq').addClass('open');
			}

	});

});
</script>

</head>

<body>

    <div id="content" class="container_24 clearfix">
      <div class="main-content">
        <div class="header-img">
          <img src="../images/header-img.jpg" alt="Bioinformatics Algorithms: An Active Learning Approach" width="960" height="330" />
        </div>
        <div class="nav">
          <ul>
            <li><a href="../index.htm">Home</a></li>
            <li><a href="../about-the-author.htm">Authors</a></li>
            <li><a href="../contents.htm">Contents</a></li>
            <li><a href="../faqs.htm">FAQs</a></li>
            <li><a href="../videos.htm">Videos & Slides</a></li>
                        <li><a href="https://secure.mybookorders.com/Orderpage/1402">Buy the Book!</a></li>
            <li><a class="active" href="../contact.htm">Contact</a></li>
          </ul>
        </div>

    	<div class="section-about">

<h1>Application Challenge: RNA-Seq</h1>



<div class="faq_container">
	<div class="faq">
		<div class="faq_question"><a name="merge_expression_values"></a>How do biologists merge expression values from different samples?</div>
		<div class="faq_answer_container">
			<div class="faq_answer">
				<p>Given experimental and control groups with multiple samples, how do biologists “‘merge”’ expression values from a given group into a single expression value? For complex reasons beyond the scope of this application challenge, we can assume that the number of reads <i>K</i><sub><i>i</i>,<i>j</i></sub> in some sample <i>j</i> that are assigned to gene <i>i</i> can be modeled by a <b>negative binomial distribution</b>.  This distribution models the number of “successes” that will occur in an indefinite sequence of identical independent trials (where each trial has probability <i>p</i> of “success” and probability 1 – <i>p</i> of “failure”) before we encounter <i>r</i> “failures”.  A simple example of a negative binomial distribution would model flipping a biased coin with a probability 0.3 of heads (“success”) and asking what the probability is of obtaining <i>k</i> heads when we reach the 10th tails.  More generally, the probability of obtaining exactly <i>k</i> successes when we encounter the <i>r</i>-th failure is</p>

				<p><br></p>

				<p><center>C(<i>k</i>+<i>r</i>-1, <i>k</i>) · (1-<i>p</i>)<sup><i>r</i></sup> · <i>p</i><sup><i>k</i></sup></center></p>

				<p><br></p>

				<p>where C(_, _) is the <a href="https://en.wikipedia.org/wiki/Combination" target="_blank">combination statistic</a>. The exact values of the parameters <i>k</i>, <i>p</i>, and <i>r</i> are unknown in advance and are estimated from the data. </p>
			</div>
		</div>
	</div>
</div>


<p><br></p>

<div class="faq_container">
	<div class="faq">
		<div class="faq_question"><a name="p_value"></a>What is a p-value?</div>
		<div class="faq_answer_container">
			<div class="faq_answer">
				<p>To define p-values, we consider two hypotheses: <b>H<sub>0</sub></b> (the <b>null hypothesis</b>) and <b>H<sub>1</sub></b> (the <b>experimental hypothesis</b>). H<sub>0</sub> states that nothing significant occurred: the results of the experiment deviated from the expected outcome purely by chance. In contrast, H<sub>1</sub> states that the results of the experiment significantly deviated from the expected outcome. In practice, we assume that the null hypothesis is true, and then ask ourselves, “What is the probability that we would obtain a result equal to or more extreme than the one obtained, given that the null hypothesis is indeed true?” This probability is what we mean by a <b>p-value</b>.</p>

				<p><br></p>

				<p>If a p-value is low, then there is a good chance that the null hypothesis is not true; as a result, we define some probability threshold for which, if the p-value of our experiment falls below this threshold, we reject H<sub>0</sub> (i.e., we conclude that the outcome was unlikely to have been caused by random chance).  Note that this does not mean that we accept the experimental hypothesis. In the case of gene expression, H<sub>0</sub> is “The measured gene expression levels between the two groups differed purely because of random chance in sampling”. H<sub>1</sub>, on the other hand, is “The gene expression levels between the two groups differed because the gene is actually being differentially expressed”. Connecting the null hypothesis to a quantifiable p-value is a complex issue, one that depends on the model used, and as a result one that can sometimes be disputed.</p>
			</div>
		</div>
	</div>
</div>

<p><br></p>

<div class="faq_container">
	<div class="faq">
		<div class="faq_question"<a name="NaN"></a>What is NaN?</div>
		<div class="faq_answer_container">
			<div class="faq_answer">
				<p>Some genes have “NaN” in place of a p-value. In computing, <b>NaN</b> (an acronym for <b>Not a Number</b>), is a numeric data type value representing an undefined value. There are three types of operations that can return NaN:</p>

				<p><br></p>

				<ul>
					<li><i>Indeterminate forms</i>: Any of the arithmetic problems that have undefined solutions. A few examples are 0/0, ∞/∞, ∞ – ∞, etc.</li>

					<li><i>Real operations with complex results</i>: Since complex numbers are not handled in floating-point calculations, they are simply stored as NaN. Two examples are the square root of a negative number and the logarithm of a negative number. </li>

					<li><i>NaN as at least one operand</i>: If you attempt to do any type of arithmetic using NaN as an operand, the result must also be NaN since arithmetic operations involving “Not a Number” values are undefined.</li>
				</ul>
			</div>
		</div>
	</div>
</div>



<!--
<div class="faq_container">
	<div class="faq">
		<div class="faq_question">Question here</div>
		<div class="faq_answer_container">
			<div class="faq_answer">
				Answer here
			</div>
		</div>
	</div>
</div>
-->



		</div>
	</div>
  <div id="footer" class="container_24 clearfix">
          <div class="footer-links">
     <img style="float: left;" src="/../images/alp_logo.png" alt="Active Learning Publishers" height="50" />  <a style="float: right;" href="http://rosalind.info/problems/list-view/?location=bioinformatics-textbook-track" target="_blank"><img style="padding:10px;"  src="/images/rosalind_logo_white.png" alt="Rosalind" height="50" /></a>
     &#169; 2018 by Phillip Compeau &amp; Pavel Pevzner | All Rights Reserved<br>
      ISBN: 978-0-9903746-3-3<br>
    </div>
  </div>
</div>


</body>
</html>
